packages/airnode-deployer/src/infrastructure/index.test.ts

```diff
@@ -866,3 +853,165 @@ describe('removeAirnode', () => {
     );
   });
 });
+
+describe('listAirnodes', () => {
+  const bucket = 'airnode-123456789';
+  const configPath = path.join(__dirname, '..', '..', 'test', 'fixtures', 'config.valid.json');
+  const directoryStructure = pick(mockBucketDirectoryStructure, [
+    '0xd0624E6C2C8A1DaEdE9Fa7E9C409167ed5F256c6',
+    '0xA30CA71Ba54E83127214D3271aEA8F5D6bD4Dace',
+  ]);
+
+  let awsGetAirnodeBucketSpy: jest.SpyInstance;
+  let awsGetBucketDirectoryStructureSpy: jest.SpyInstance;
+  let awsGetFileFromBucketSpy: jest.SpyInstance;
+  let gcpGetAirnodeBucketSpy: jest.SpyInstance;
+  let gcpGetBucketDirectoryStructureSpy: jest.SpyInstance;
+  let gcpGetFileFromBucketSpy: jest.SpyInstance;
+  let consoleSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    awsGetAirnodeBucketSpy = jest.spyOn(aws, 'getAirnodeBucket').mockImplementation(() => Promise.resolve(bucket));
+    awsGetBucketDirectoryStructureSpy = jest
+      .spyOn(aws, 'getBucketDirectoryStructure')
+      .mockImplementation(() => Promise.resolve(directoryStructure));
+    awsGetFileFromBucketSpy = jest
+      .spyOn(aws, 'getFileFromBucket')
+      .mockImplementation(() => Promise.resolve(fs.readFileSync(configPath).toString()));
+    gcpGetAirnodeBucketSpy = jest.spyOn(gcp, 'getAirnodeBucket').mockImplementation(() => Promise.resolve(bucket));
+    gcpGetBucketDirectoryStructureSpy = jest
+      .spyOn(gcp, 'getBucketDirectoryStructure')
+      .mockImplementation(() => Promise.resolve(directoryStructure));
+    gcpGetFileFromBucketSpy = jest
+      .spyOn(gcp, 'getFileFromBucket')
+      .mockImplementation(() => Promise.resolve(fs.readFileSync(configPath).toString()));
```

Comment:
Nit: You can simplify this using https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvaluevalue